lhotse/recipes/mtedx.py

```diff
@@ -0,0 +1,237 @@
+"""
+MTEDx is a collection of transcribed and translated speech corpora:
+ https://openslr.org/100
+
+It has 8 languages:
+es - 189h
+fr - 189h
+pt - 164h
+it - 107h
+ru - 53h
+el - 30h
+ar - 18h
+de - 15h
+
+A subset of this audio is translated and split into the following partitions:
+     - train
+     - dev
+     - test
+     - iwslt2021 sets
+
+This recipe only prepares the ASR portion of the data.
+"""
+import logging
+import re
+from collections import defaultdict
+from typing import Dict, Iterable, List, Optional, Union
+from pathlib import Path
+
+from cytoolz import sliding_window
+
+import tarfile
+import shutil
+from lhotse import Recording, RecordingSet, SupervisionSegment, SupervisionSet, validate_recordings_and_supervisions
+from lhotse.qa import remove_missing_recordings_and_supervisions, trim_supervisions_to_recordings
+from lhotse.utils import Pathlike, urlretrieve_progress
+import regex as re2
+from functools import partial
+import unicodedata
+
+
+# Keep Markings such as vowel signs, all letters, and decimal numbers 
+VALID_CATEGORIES = ('Mc', 'Mn', 'Ll', 'Lm', 'Lo', 'Lt', 'Lu', 'Nd', 'Zs')
+noise_pattern = re2.compile(r'\([^)]*\)', re2.UNICODE)
+apostrophe_pattern = re2.compile(r"(\w)'(\w)")
+html_tags = re2.compile(r"(&[^ ;]*;)|(</?[iu]>)")
+KEEP_LIST = [u'\u2019']
+
+
+ASR = ('es', 'fr', 'pt', 'it', 'ru', 'el', 'ar', 'de',)
+
+
+ISOCODE2LANG = {
+    'fr': 'French',
+    'es': 'Spanish',
+    'pt': 'Portuguese',
+    'it': 'Italian',
+    'ru': 'Russian',
+    'el': 'Greek',
+    'ar': 'Arabic',
+    'de': 'German',
+}
+
+
+def download_and_untar(
+    target_dir: Pathlike = '.',
+    force_download: Optional[bool] = False
+    language: str = 'fr',
```

Comment:
could we change this to accept either:
- a list of languages; or
- a single language string; or
- a string "all" (default)?